### PR TITLE
add Pkg to test env as a dep rather than via LOAD_PATH

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaLang/julia/issues/51820

### CODE REVIEW

File: src/Operations.jl [hunk 1]
```diff
@@ -1853,6 +1851,7 @@ function gen_target_project(ctx::Context, pkg::PackageSpec, source_path::String,
     if projectfile_path(source_path; strict=true) === nothing
         # no project file, assuming this is an old REQUIRE package
         test_project.deps = copy(env.manifest[pkg.uuid].deps)
+        get!(test_project.deps, "Pkg", PkgUUID) # so that the test environment can be instantiated & precompiled
```
Comment (IanButterworth) [inline src/Operations.jl:1854]: Perhaps to catch the edge case where Pkg is another package..
```suggestion
        # add Pkg so that the test environment sandbox subprocesses can be precompiled
        if get!(test_project.deps, "Pkg", PkgUUID) != PkgUUID
            @warn """
            A package called Pkg is declared with a UUID that doesn't match the Pkg stdlib.
            This may cause unexpected behavior"""
        end
```
